spec/services/claim_fast_tracking/max_rating_annotator_spec.rb

```diff
@@ -22,10 +24,67 @@
         ]
       end
 
-      it 'mutates just the tinnitus disability with a max rating' do
-        subject
+      it 'does not mutate any disability with a max rating' do
         max_ratings = disabilities_response.rated_disabilities.map(&:maximum_rating_percentage)
-        expect(max_ratings).to eq([nil, 10, nil])
+        expect(max_ratings).to eq([nil, nil, nil])
+      end
+    end
+
+    context 'with disability_526_maximum_rating enabled' do
+      before { Flipper.enable(:disability_526_maximum_rating_api) }
+      after { Flipper.disable(:disability_526_maximum_rating_api) }
+
+      context 'when a disabilities response contains rating for a disability' do
+        let(:disabilities_data) do
+          [
+            { name: 'Hypertension', diagnostic_code: 7101, rating_percentage: 20 },
+            { name: 'Tinnitus', diagnostic_code: 6260, rating_percentage: 10 },
+            { name: 'Vertigo', diagnostic_code: 6204, rating_percentage: 30 }
+          ]
+        end
+
+        it 'mutates just the rated disability with a max rating' do
+          VCR.use_cassette('virtual_regional_office/max_ratings') do
+            subject
+            max_ratings = disabilities_response.rated_disabilities.map(&:maximum_rating_percentage)
+            expect(max_ratings).to eq([nil, 10, nil])
+          end
+        end
+      end
+
+      context 'when a disabilities response does not contains rating any disability' do
+        let(:disabilities_data) do
+          [
+            { name: 'Hypertension', diagnostic_code: 7101, rating_percentage: 20 },
+            { name: 'Vertigo', diagnostic_code: 6204, rating_percentage: 30 }
+          ]
+        end
+
+        it 'mutates none of the disabilities with a max rating' do
+          VCR.use_cassette('virtual_regional_office/max_ratings') do
+            subject
+            max_ratings = disabilities_response.rated_disabilities.map(&:maximum_rating_percentage)
+            expect(max_ratings).to eq([nil, nil])
+          end
+        end
+      end
+
+      context 'when max rating VRO endpoint fails' do
+        let(:disabilities_data) do
+          [
+            { name: 'Hypertension', diagnostic_code: 7101, rating_percentage: 20 },
+            { name: 'Tinnitus', diagnostic_code: 6260, rating_percentage: 10 },
+            { name: 'Vertigo', diagnostic_code: 6204, rating_percentage: 30 }
```

Comment:
This data setup is getting a little repeated, and the neat thing about RSpec is that you can move a `let` into a broader context and all nested contexts will receive that value. Then, if you have a more specific context that needs a different value (e.g. the no tinnitus example) you can override it with another `let` within the more specific context.